### PR TITLE
Feature/add api key authentication

### PR DESCRIPTION
**Pull request by brim-borium**

### CODE REVIEW

File: android/src/main/kotlin/de/minimalme/giphy_sdk/GiphySdkPlugin.kt [hunk 1]
```diff
@@ -71,25 +80,107 @@ class GiphySdkPlugin: FlutterPlugin, MethodCallHandler, ActivityAware {
...
+        GPHContentType.gif,
+        GPHContentType.recents,
+        GPHContentType.sticker,
+        GPHContentType.text),
+      selectedContentType = GPHContentType.gif,
+      showSuggestionsBar = true,
+      showConfirmationScreen= true,
+      enableDynamicText = true )
+
+    if(!giphySettings.isNullOrEmpty()){
```
Comment (samtun) [inline android/src/main/kotlin/de/minimalme/giphy_sdk/GiphySdkPlugin.kt]: add spacing before and after ()

File: android/src/main/kotlin/de/minimalme/giphy_sdk/GiphySdkPlugin.kt [hunk 2]
```diff
@@ -71,25 +80,107 @@ class GiphySdkPlugin: FlutterPlugin, MethodCallHandler, ActivityAware {
...
+          GridType.waterfall,
+          theme = mappedTheme,
+          stickerColumnCount = columnCount,
+          mediaTypeConfig = mediaTypeConfig,
+          selectedContentType = mappedSelectedContentType,
+          showSuggestionsBar = showSuggestionsBar,
+          showConfirmationScreen= showConfirmationScreen,
+          enableDynamicText = enabledDynamicText )
+      }
+     catch(ex: Exception) {
```
Comment (samtun) [inline android/src/main/kotlin/de/minimalme/giphy_sdk/GiphySdkPlugin.kt]: Indentation is somewhat off here (maybe just reformat the file)

File: example/pubspec.yaml [hunk 3]
```diff
@@ -53,7 +54,8 @@ flutter:
   uses-material-design: true
 
   # To add assets to your application, add an assets section, like this:
```
Comment (samtun) [inline example/pubspec.yaml]: Maybe remove these default comments (also the images below)

File: lib/models/giphy_settings.dart [hunk 4]
```diff
@@ -0,0 +1,99 @@
+import 'package:giphy_sdk/enums/giphy_content_type.dart';
+import 'package:giphy_sdk/enums/giphy_theme.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'giphy_settings.g.dart';
+
+/// Used to configure the apperance and configuration of the giphy selection.
+///
+/// It uses a default set of settings but can indivudally be changed. For ease
```
Comment (samtun) [inline lib/models/giphy_settings.dart]: individually

File: lib/models/user_dictionary.dart [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'user_dictionary.g.dart';
+
+@JsonSerializable()
+class UserDictionary {
+  @JsonKey(name: 'rk')
```
Comment (samtun) [inline lib/models/user_dictionary.dart:7]: Suggestion: Add comments here that explain what those values are

File: test/giphy_sdk_test.dart [hunk 6]
```diff
@@ -18,6 +18,6 @@ void main() {
   });
 
   test('getPlatformVersion', () async {
-    expect(await GiphySdk.connectToGiphy(apiKey: ""), true);
+    expect(await GiphySdk.openGiphySelection(apiKey: ""), true);
```
Comment (samtun) [inline test/giphy_sdk_test.dart:21]: I guess this test was changed by automated refactoring? Maybe add actual tests here if possible

File: lib/models/user_dictionary.dart (hunk 5 above)
Comment (brim-borium) [reply lib/models/user_dictionary.dart:7]: I don't know yet ^^

File: test/giphy_sdk_test.dart (hunk 6 above)
Comment (brim-borium) [reply test/giphy_sdk_test.dart:21]: Would leave as is

**Review by samtun — APPROVED** [review]
Code looks good. Tested on Android 8.1
Crash due to different fields being used in gif / text types is ok for now.

**Pull request merged**